nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -272,18 +316,38 @@ class BtActionNode : public BT::ActionNodeBase
         }
       };
 
-    auto future_goal_handle = action_client_->async_send_goal(goal_, send_goal_options);
+    goal_handle_.reset();
+    future_goal_handle_ = action_client_->async_send_goal(goal_, send_goal_options);
+    time_goal_sent_ = node_->now();
+    time_elapsed_since_goal_sent_ = std::chrono::milliseconds(0);
```

Comment:
I'd argue sending the `elapsed_time` as a function input is easier to logically track, but this also works